t/data/OpenQA/Plugin/WebAPI/Hello.pm

```diff
@@ -0,0 +1,36 @@
+package OpenQA::Plugin::WebAPI::Hello;
+use Mojo::Base 'Mojolicious::Plugin';
+use Cwd 'abs_path';
+use File::Basename;
+
+# my $script_dirname = dirname(abs_path(__FILE__));
+my $script_dirname = dirname(__FILE__);
+my $name;
+
+sub register {
+    my ($self, $app, $config) = @_;
+    my $prefix = $config->{route} // $app->routes->any('/plugin');
+    $prefix->to(return_to => $config->{return_to} // '/');
+
+    $name = $config->{name} // "";
+
+    # Templates
+    push @{$app->renderer->paths}, $script_dirname;
```

Comment:
Don't mix templates and packages directly, use at least a templates directory. Personally, i prefer the convention to use a `resources/templates` directory, like the [Minion admin ui](https://github.com/mojolicious/minion/tree/master/lib/Mojolicious/Plugin/Minion/resources) does.